app/src/main/java/pub/devrel/easypermissions/sample/MainActivity.java

```diff
@@ -63,14 +63,35 @@ public void onClick(View v) {
     public void onActivityResult(int requestCode, int resultCode, Intent data) {
         super.onActivityResult(requestCode, resultCode, data);
 
+        // Do something after user returned from app settings screen. User may be
+        // changed/updated the permissions. In this sample, we could check whether the user has
+        // changed/updated permission for camera, access fine location, or read contacts.
         if (requestCode == EasyPermissions.SETTINGS_REQ_CODE) {
-            // Do something after user returned from app settings screen
-            // Let's show Toast for example
-            Toast.makeText(this, R.string.returned_from_app_settings_to_activity, Toast.LENGTH_SHORT)
+            // Check the changed/updated permissions manually. Let's show toast.
+            Toast.makeText(this, getUpdatedPermissionsText(), Toast.LENGTH_SHORT)
                     .show();
         }
     }
 
+    private String getUpdatedPermissionsText() {
```

Comment:
While I think this is a nice "correct" way to do this I want `MainActivity` to be as concise as possible.  Let's just revert back to what you had before, where it just toasts a string saying the user came back from the Settings screen.   Then we can drop this whole function.  Advanced users will be able to figure out what to do here.